src/components/sketches/Scene5_happy.ts

```diff
@@ -0,0 +1,266 @@
+    if (ringParticles.length > MAX_RING_PARTICLES) {
+      // Delete old ringParticles (from beginning of array)
+      ringParticles.splice(0, p.floor((ringParticles.length - MAX_RING_PARTICLES * 0.8) * EASING));
+    }
+
+    // Loop through the array and show each particle
+    for (let i = 0; i < ringParticles.length; i++) {
+      ringParticles[i].update();
+      if (ringParticles[i].delete()) {
+        ringParticles.splice(i, 1);
+      } else {
+        ringParticles[i].show();
+      }
+    }
+  };
+  // Spawn ringParticles in ring with centre mouse x and y coords
+  p.mouseClicked = () => {
+    for (let i = 0; i < RING_COVERAGE; i += 4) {
+      const c = new RingParticle(p.mouseX, p.mouseY, i);
+      ringParticles.push(c);
+    }
+  };
+
+  class Firework {
+    public explodeSize: number;
+    public exploded: boolean;
+    private _firingParticle: Particle;
+    private _particles: Array<Particle>;
+
+    constructor() {
+      this.explodeSize = p.random(0, EXPLODE_RANGE);
+      this._firingParticle = new Particle(p.random(p.width), p.height, true);
+      this.exploded = false;
+      this._particles = [];
+    }
+
+    update() {
+      if (!this.exploded) {
+        this._firingParticle.update(gravity);
+
+        if (this._firingParticle.vel.y >= 0) {
+          this.explode();
+        }
+      }
+
+      for (let i = this._particles.length - 1; i >= 0; i--) {
+        this._particles[i].update(gravity);
+
+        if (this._particles[i].done()) {
+          this._particles.splice(i, 1);
+        }
+      }
+    }
+
+    done() {
+      return (this.exploded && this._particles.length === 0);
+    }
+
+    show() {
+      if (!this.exploded) {
+        this._firingParticle.show();
+      }
+
+      for (let i = 0; i < this._particles.length; i++) {
+        this._particles[i].show();
+      }
+    }
+
+    explode() {
+      this.exploded = true;
+      for (let i = 0; i < this.explodeSize * EXPLODE_OFFSET; i++) {
+        const particle = new Particle(
+          this._firingParticle.pos.x,
+          this._firingParticle.pos.y,
+          false,
+        );
+        this._particles.push(particle);
+      }
+    }
+  }
+
+  class Particle {
+    pos: p5.Vector;
+    public isFirework: boolean;
+    lifespan: number;
+    private _acc: p5.Vector;
+    rand: number;
+    vel: p5.Vector;
+
+    constructor(x: number, y: number, firework: boolean) {
+      this.pos = p.createVector(x, y);
+      this.isFirework = firework;
+      this.lifespan = FIREWORK_LIFESPAN;
+      this._acc = p.createVector(0, 0);
+      this.rand = p.random(0, 3);
+      this.vel = this.isFirework
+        ? p.createVector(0, p.random(-12, -8))
+        : p5.Vector.random2D().mult(p.random(2, 10));
+    }
+
+    update(force: p5.Vector) {
+      if (!this.isFirework) {
+        this.vel.mult(0.9);
+        this.lifespan -= 4;
+      }
+      this._acc.add(force);
+      this.vel.add(this._acc);
+      this.pos.add(this.vel);
+      this._acc.mult(0);
```

Comment:
So my point here is:

at line 152: `this._acc = p.createVector(0,0)`
We start with an empty accerlation vector.

Then on every update, at line 164: `this._acc.add(force)`
We add to a zeroed acceleration vector, `force`

Then in the same update, at line 167: `this._acc.mult(0)`
We set acceleration back to zero

Why do we need any of this? Shouldn't we just add force to `this.vel`

Reply:
i see what u mean. good catch, deleting _acc field.